### PR TITLE
Issue 882 - core-metadata Command: Eliminate Mongo Types from Models

### PR DESCRIPTION
**Pull request by michaelestrin**
#882 

Signed-off-by: Michael Estrin <m.estrin@dell.com>

### CODE REVIEW

**Comment by michaelestrin** [comment]
Requires https://github.com/edgexfoundry/blackbox-testing/issues/114

**Comment by michaelestrin** [comment]
Requires https://github.com/edgexfoundry/blackbox-testing/pull/115

**Comment by tsconn23** [comment]
Testing this PR using blackbox PR https://github.com/edgexfoundry/blackbox-testing/pull/115

Please re-run the blackbox tests for command_error_4xx

- Test 340 fails to return the correct status code
- Test 349 causes a panic in the service

```
2018/12/13 12:49:47 http: panic serving [::1]:61276: runtime error: invalid memory address or nil pointer dereference
goroutine 52 [running]:
net/http.(*conn).serve.func1(0xc0003b80a0)
	/usr/local/go/src/net/http/server.go:1746 +0xd0
panic(0x145e260, 0x1845e60)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/models.(*Get).ToContract(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/models/get.go:28 +0x5e
github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/models.(*Command).ToContract(0xc0002505a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/models/command.go:44 +0x91
github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo.(*MongoClient).GetCommandById(0xc000150230, 0xc00028c500, 0xc0000c2480, 0x18, 0x0, 0x0)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/metadata.go:886 +0x338
github.com/edgexfoundry/edgex-go/internal/core/metadata.restUpdateCommand(0x1564c40, 0xc0004881c0, 0xc0003be500)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/internal/core/metadata/rest_command.go:79 +0x2c8
net/http.HandlerFunc.ServeHTTP(0x14fc728, 0x1564c40, 0xc0004881c0, 0xc0003be500)
	/usr/local/go/src/net/http/server.go:1964 +0x44
github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0000c0410, 0x1564c40, 0xc0004881c0, 0xc0003be500)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/mux/mux.go:114 +0xe0
github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/context.ClearHandler.func1(0x1564c40, 0xc0004881c0, 0xc000178c00)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/context/context.go:141 +0x8b
net/http.HandlerFunc.ServeHTTP(0xc00036d480, 0x1564c40, 0xc0004881c0, 0xc000178c00)
	/usr/local/go/src/net/http/server.go:1964 +0x44
net/http.serverHandler.ServeHTTP(0xc0000a1860, 0x1564c40, 0xc0004881c0, 0xc000178c00)
	/usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc0003b80a0, 0x1565240, 0xc0001a6180)
	/usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

**Comment by michaelestrin** [comment]
I hadn't run those.  Chalk it up to learning curve friction.  Will rework and update PR.

**Comment by michaelestrin** [comment]
@tsconn23 - both command and command_error_4xx blackbox tests go green now.

File: internal/pkg/db/mongo/models/command.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * the License.
+ *******************************************************************************/
+
+package models
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+	contract "github.com/edgexfoundry/edgex-go/pkg/models"
+	"github.com/google/uuid"
+	"gopkg.in/mgo.v2/bson"
```
Comment (tsconn23) [inline internal/pkg/db/mongo/models/command.go]: Need to update this import with new globalsign provider

Comment (michaelestrin) [reply internal/pkg/db/mongo/models/command.go]: Good catch.  Done.

**Pull request merged**